lms/djangoapps/verify_student/models.py

```diff
@@ -1186,3 +1186,64 @@ def check_user_skipped_reverification_exists(cls, user, course_id):
             Boolean
         """
         return cls.objects.filter(user=user, course_id=course_id).exists()
+
+
+class CreditCourse(models.Model):
+    """Model for tracking the credit course."""
+
+    course_key = CourseKeyField(max_length=255, db_index=True, unique=True)
+
+
+class CreditProvider(models.Model):
+    """This model represents an institution that can grant credit for a course.
+    each provider is identified by unique id e.g ASU
+    """
+    provide_id = models.CharField(max_length=255, db_index=True, unique=True)
+    display_name = models.CharField(max_length=255)
+    created_at = models.DateTimeField(auto_now_add=True)
+    modified_at = models.DateTimeField(auto_now=True)
+
+
+class CreditRequirement(models.Model):
+    """This model represents the credit requirments.each requirments is
+    uniquely identified by a “namespace” and a “name” and course id.
+    “configuration” will be dictionary, the format of which varies by the type
+    of requirement.
+    """
+    course = models.ForeignKey(CreditCourse, related_name="credit_requirement_course")
+    requirement_namespace = models.CharField(max_length=255)
+    requirement_name = models.CharField(max_length=255)
+    configuration = models.TextField()
```

Comment:
According to @wedaly's document, `configuration` will be JSON-encoded. I'd suggest using a `JSONField()` here instead of a text field: https://github.com/bradjasper/django-jsonfield.